src/impls/core_/num.rs

```diff
@@ -0,0 +1,26 @@
+use core::num;
+
+use super::*;
+
+macro_rules! non_zero {
+    ($type:ty) => {
+        impl Format for $type {
+            fn format(&self, fmt: Formatter) {
+                crate::write!(fmt, "{}", self.get());
```

Comment:
This format string could specify the type directly, which would be more efficient. Not a blocker, but please create an issue tracking this (or add a FIXME) if you don't want to do it in this PR.

Reply:
I see. I will try to do it here.

Reply:
What do you think?

Reply:
perfect!

Reply:
😊